From 572ed8a8ce8aa0ce4cf23303291c873d11ac4470 Mon Sep 17 00:00:00 2001
From: Tobias <ruck.tobias@gmail.com>
Date: Mon, 27 Jun 2022 01:53:09 +0200
Subject: [PATCH 167/196] Activate testnet 21 days before mainnet (#256)

This allows to simulate activation before and make sure everything works properly.
---
 src/chainparams.cpp | 22 ++++++++++++++++------
 1 file changed, 16 insertions(+), 6 deletions(-)

diff --git a/src/chainparams.cpp b/src/chainparams.cpp
index afeaef625..19293dfea 100644
--- a/src/chainparams.cpp
+++ b/src/chainparams.cpp
@@ -232,12 +232,18 @@ public:
         consensus.defaultAssumeValid =
             ChainParamsConstants::TESTNET_DEFAULT_ASSUME_VALID;
 
+        // Testnet activation times are 21 days before mainnet
+        Consensus::Params mainnetConsensus = CMainParams().GetConsensus();
+        const int testnetActivationOffset = 21 * 24 * 3600;
         // 2021-12-21T15:59:00.000Z protocol upgrade
-        consensus.exodusActivationTime = 1640102340;
+        consensus.exodusActivationTime =
+            mainnetConsensus.exodusActivationTime - testnetActivationOffset;
         // 2022-06-21T09:14:00.000Z protocol upgrade
-        consensus.leviticusActivationTime = 1655802840;
+        consensus.leviticusActivationTime =
+            mainnetConsensus.leviticusActivationTime - testnetActivationOffset;
         // 2022-12-12T21:48:00.000Z protocol upgrade
-        consensus.numbersActivationTime = 1670881680;
+        consensus.numbersActivationTime =
+            mainnetConsensus.numbersActivationTime - testnetActivationOffset;
 
         // "ltdk" with MSB set
         diskMagic[0] = 0xec;
@@ -335,12 +341,16 @@ public:
         // valid.
         consensus.defaultAssumeValid = BlockHash();
 
+        // Regtest activation times are the same as mainnet
+        Consensus::Params mainnetConsensus = CMainParams().GetConsensus();
         // 2021-12-21T15:59:00.000Z protocol upgrade
-        consensus.exodusActivationTime = 1640102340;
+        consensus.exodusActivationTime = mainnetConsensus.exodusActivationTime;
         // 2022-06-21T09:14:00.000Z protocol upgrade
-        consensus.leviticusActivationTime = 1655802840;
+        consensus.leviticusActivationTime =
+            mainnetConsensus.leviticusActivationTime;
         // 2022-12-12T21:48:00.000Z protocol upgrade
-        consensus.numbersActivationTime = 1670881680;
+        consensus.numbersActivationTime =
+            mainnetConsensus.numbersActivationTime;
 
         // "lrdk" with MSB set
         diskMagic[0] = 0xec;
-- 
2.39.5 (Apple Git-154)

